ape_hardhat/__init__.py

```diff
@@ -0,0 +1,237 @@
+try:
+    from importlib.metadata import PackageNotFoundError as _PackageNotFoundError  # type: ignore
+    from importlib.metadata import version as _version  # type: ignore
+except ModuleNotFoundError:
+    from importlib_metadata import PackageNotFoundError as _PackageNotFoundError  # type: ignore
+    from importlib_metadata import version as _version  # type: ignore
+
+try:
+    __version__ = _version(__name__)
+except _PackageNotFoundError:
+    # package is not installed
+    __version__ = "<unknown>"
+
+import atexit
+import ctypes
+import platform
+import random
+import signal
+import sys
+import time
+from subprocess import PIPE, Popen, call
+from typing import Any, List, Optional
+
+from ape import plugins
+from ape_http.providers import DEFAULT_SETTINGS, EthereumProvider, NetworkConfig
+
+EPHEMERAL_PORTS_START = 49152
+EPHEMERAL_PORTS_END = 60999
+HARDHAT_CHAIN_ID = 31337
+HARDHAT_CONFIG = """
+// See https://hardhat.org/config/ for config options.
+module.exports = {
+};
+"""
+HARDHAT_START_NETWORK_RETRIES = [0.1, 0.2, 0.3, 0.5, 1.0]  # seconds between network retries
+HARDHAT_START_PROCESS_ATTEMPTS = 3  # number of attempts to start subprocess before giving up
+
+
+def _signal_handler(signum, frame):
+    """Runs on SIGTERM and SIGINT to force ``atexit`` handlers to run."""
+    atexit._run_exitfuncs()
+    sys.exit(0)
```

Comment:
Is `0` the right exit code here? The standard is to do `130`.

```suggestion
    sys.exit(130)
```

Reply:
Looks like getting killed via SIGTERM is 143, via SIGINT or Ctrl+C is 130, and via SIGQUIT is 131:

```
$ python3
Python 3.8.10 (default, Jun  2 2021, 10:49:15)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.system("kill {}".format(os.getpid()))
Terminated

$ echo $?
143

---

$ cat x.py
input("hey...")

cuboid@gts ~
$ python3 x.py
hey...^CTraceback (most recent call last):
  File "x.py", line 1, in <module>
    input("hey...")
KeyboardInterrupt
# (this comes from hitting Ctrl+C)

$ echo $?
130

---

$ python3 x.py
hey...Quit (core dumped)
# (killed the process via `ps auxwww | grep "python3 x.p[y]" | awk '{print $2}' | xargs kill -s QUIT` in another terminal)

$ echo $?
131
```

I added a conditional to return 143 or 130 based on the `signum`.